dev-util/maturin/maturin-0.13.0.ebuild

```diff
@@ -0,0 +1,419 @@
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+	windows-sys-0.36.1
+	windows_aarch64_msvc-0.36.1
+	windows_i686_gnu-0.36.1
+	windows_i686_msvc-0.36.1
+	windows_x86_64_gnu-0.36.1
+	windows_x86_64_msvc-0.36.1
+	xattr-0.2.3
+	xwin-0.2.5
+	zbus-1.9.3
+	zbus_macros-1.9.3
+	zip-0.6.2
+	zvariant-2.10.0
+	zvariant_derive-2.10.0"
+CRATES_TEST="
+	libc-0.2.119
+	libc-0.2.125
+	once_cell-1.10.0
+	once_cell-1.11.0
+	once_cell-1.12.0
+	once_cell-1.9.0
+	parking_lot-0.12.0
+	proc-macro2-1.0.38
+	proc-macro2-1.0.39
+	pyo3-0.16.5
+	pyo3-build-config-0.16.0
+	pyo3-build-config-0.16.5
+	pyo3-ffi-0.16.0
+	pyo3-ffi-0.16.5
+	pyo3-macros-0.16.5
+	pyo3-macros-backend-0.16.5
+	python3-dll-a-0.2.3
+	quote-1.0.18
+	smallvec-1.8.0
+	syn-1.0.94
+	syn-1.0.95
+	target-lexicon-0.12.3
+	unicode-ident-1.0.0
+	unicode-xid-0.2.3
+	unindent-0.1.9"
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( pypy3 python3_{8..11} )
+inherit cargo distutils-r1 flag-o-matic
+
+DESCRIPTION="Build and publish crates with pyo3, rust-cpython and cffi bindings"
+HOMEPAGE="https://maturin.rs/"
+SRC_URI="
+	https://github.com/PyO3/maturin/archive/refs/tags/v${PV}.tar.gz -> ${P}.gh.tar.gz
+	$(cargo_crate_uris)
+	test? ( $(cargo_crate_uris ${CRATES_TEST}) )"
+
+LICENSE="
+	0BSD Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD CC0-1.0 ISC MIT MPL-2.0 openssl
+	doc? ( CC-BY-4.0 OFL-1.1 )"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~riscv ~x86"
+IUSE="doc test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	$(python_gen_cond_dep '
+		dev-python/tomli[${PYTHON_USEDEP}]
+	' 3.{8..10} pypy3)"
+BDEPEND="
+	>=dev-python/setuptools-rust-1.4[${PYTHON_USEDEP}]
+	doc? ( app-text/mdbook )
+	test? (
+		${RDEPEND}
+		$(python_gen_cond_dep '
+			dev-python/cffi[${PYTHON_USEDEP}]
+		' 'python*')
+		dev-python/boltons[${PYTHON_USEDEP}]
+		dev-python/virtualenv[${PYTHON_USEDEP}]
+	)"
+
+QA_FLAGS_IGNORED=".*/${PN}"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.12.8-zig-tests.patch
+)
+
+src_prepare() {
+	distutils-r1_src_prepare
+
+	# used to prevent use of network during tests
+	cat > "${T}"/pip.conf <<-EOF || die
+		[install]
+		no-index = yes
+		no-dependencies = yes
+	EOF
+
+	# TODO: package-agnostic way to handle IUSE=debug with setuptools-rust?
+	use !debug || sed -i "s/^cargo_args = \[/&'--profile','dev',/" setup.py || die
+
+	filter-lto # undefined references with ring crate
+}
+
+python_compile_all() {
+	use !doc || mdbook build -d html guide || die
+}
+
+src_test() {
+	mv test-crates{,.orig} || die
+	distutils-r1_src_test
+}
+
+python_test() {
+	local -x PIP_CONFIG_FILE=${T}/pip.conf
+	local -x VIRTUALENV_SYSTEM_SITE_PACKAGES=1
+
+	local skip=(
+		--skip locked_doesnt_build_without_cargo_lock
+		# move below when >=rust-1.62 is stable, xfail "pass" with 1.60-1.61
+		--skip pyo3_no_extension_module
+	)
+	[[ ${EPYTHON} == pypy3 ]] && skip+=(
+		# test enables pyo3's auto-initialize that is incompatible with pypy
+		--skip integration_pyo3_bin
+		# wants the missing libpypy*-c.so
+#		--skip pyo3_no_extension_module
```

Comment:
Is there a reason to keep it if it's commented out?

Reply:
It's just a self-note to uncomment it when drop it above.